### PR TITLE
JDK-8289510 : Improve test coverage for XPath Axes: namespace

### PR DESCRIPTION
**Pull request by mahendrachhipa**
Added tests for namespace Axis. Tests incluse namspace expressions, namespace node counts and namespace node scope.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289510](https://bugs.openjdk.org/browse/JDK-8289510): Improve test coverage for XPath Axes: namespace

### Reviewers
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * @bwhuang-us (no known github.com user name / role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9906/head:pull/9906` \
`$ git checkout pull/9906`

Update a local copy of the PR: \
`$ git checkout pull/9906` \
`$ git pull https://git.openjdk.org/jdk pull/9906/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9906`

View PR using the GUI difftool: \
`$ git pr show -t 9906`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9906.diff">https://git.openjdk.org/jdk/pull/9906.diff</a>

</details>

### CODE REVIEW

**Comment by mahendrachhipa** [comment]
Hi @JoeWang-Java , Could you please review this PR.

File: test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpNamespaceTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    }
+
+    @DataProvider(name = "namespaceXpathNodeCount")
+    public Object[][] getNamespaceXpathExpressionNodeCount() {
+        return new Object[][] {
+                {"/Customers/namespace::*", 2},
+                {"/Customers/Customer/namespace::*", 3},
+                {"/Customers/Customer/Address/namespace::*", 2},
+                {"/Customers/Customer/Address/namespace::*", 2},
```
Comment (JoeWang-Java) [inline test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpNamespaceTest.java:127]: Duplicate test cases.

**Review by JoeWang-Java — COMMENTED** [review]
Thanks Mahendra! 

In general, it would be good to add javadoc/notes to each test and data provider.

The test cases themselves look okay, although they are a bit repetitive in that they are all in the form of a forward axis starting from the root. In practical use cases, it's most useful to perform node test with various namespace-related functions in order to locate a node or nodeset. Namespace changes the way how a node or nodeset is located, e.g. /a/b/c finds c, but as soon as namespace(s) are added, that same xpath expression no longer finds it. I remember sub-task JDK-8289948 (https://github.com/openjdk/jdk/pull/9633) had some of that. It would be good to discuss with Bill @bwhuang-us  or have Bill review this PR, as we talked about more comprehensive test coverage for namespaces in this sub-task.

**Comment by bwhuang-us** [comment]
One last thing I'd like to add is the default namespace, notated as xmlns="". Since this will impact most of the existing test cases, I am fine with adding it to one of the grandchildren and testing from there.

**Review by bwhuang-us — CHANGES_REQUESTED** [review]

**Review by bwhuang-us — APPROVED** [review]

**Pull request closed without merging**